raccoochat/RaccooChatDB.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "gen-cpp/RaccooChat.h"
+
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace raccoochat {
+
+class RaccooChatDB {
```

Comment:
lets have this class as an abstract class, with virtual methods

like
  virtual void F(...) = 0;

then you can derive a few child classes, depending on the implementation, like
class FileDB: public RaccooDB  or
class MysqlDB: public RaccooDB